packages/assistify-ai/server/methods/SmartiReload.js

```diff
@@ -58,35 +64,41 @@ Meteor.methods({
 				}
 			});
 		SystemLogger.debug('Room Id: ', rid, ' is out of sync');
-	}
-});
+	},
 
-Meteor.methods({
-	tryResync(rid) {
+	tryResync(rid, force) {
 		SystemLogger.debug('Sync all unsynced messages in room: ', rid);
 		const room = RocketChat.models.Rooms.findOneById(rid);
 		const messageDB = RocketChat.models.Messages;
 		const unsync = messageDB.find({ lastSync: { $exists: false }, rid, t: { $exists: false } }).fetch();
-		if (unsync.length === 0 && !room.outOfSync) {
-			SystemLogger.debug('Room is already in sync');
-			return true;
-		} else {
-			SystemLogger.debug('Messages out of sync: ', unsync.length);
+		if (!force || force !== true) {
+			if (unsync.length === 0 && !room.outOfSync) {
+				SystemLogger.debug('Room is already in sync');
+				return true;
+			} else {
+				SystemLogger.debug('Messages out of sync: ', unsync.length);
+			}
 		}
 		let conversation;
 
 		// conversation exists for channel?
 		SystemLogger.debug('Smarti - Trying legacy service to retrieve conversation ID...');
-		conversation = SmartiProxy.propagateToSmarti(verbs.get,
-			`legacy/rocket.chat?channel_id=${ rid }`, null, (error) => {
-				// 404 is expected if no mapping exists
-				if (typeof error.response === 'undefined') {
-					return null;
-				}
-				if (error.response.statusCode === 404) {
-					return false;
+
+		conversation = RocketChat.RateLimiter.limitFunction(
+			SmartiProxy.propagateToSmarti, 5, 1000, {
```

Comment:
We should limit the rate of this method being called - but not the interactions inside this method with Smarti - it would slow-down the sync.